traits/ctraits.c

```diff
@@ -2911,7 +2911,7 @@ trait_new(PyTypeObject *trait_type, PyObject *args, PyObject *kw)
     }
 
     if ((kind >= 0) && (kind <= 8)) {
-        trait = (trait_object *)PyType_GenericNew(trait_type, args, kw);
+        trait = (PyObject *)PyType_GenericNew(trait_type, args, kw);
```

Comment:
You want to do the cast at the `return trait` line. `trait-getattr` and `trait-setattr` won't make sense for a general `PyObject *`.